### PR TITLE
[apmazure] storage queue

### PR DESCRIPTION
**Pull request by stuartnelson3**
Add azure queue storage instrumentation.

There are still 2 other azure storage services to implement before #885 is officially "done":

- Table
- File Storage

### CODE REVIEW

**Review by axw — COMMENTED** [review]
Looks great! Just one issue about testing of the queue instrumentation, otherwise LGTM. I didn't verify the spec is being adhered to exactly, I'll take your word on that.

File: module/apmazure/queue_test.go [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+	"go.elastic.co/apm/model"
+	"go.elastic.co/apm/transport/transporttest"
+)
+
+func TestQueueSend(t *testing.T) {
+	retry := azqueue.RetryOptions{
+		MaxTries: 1,
+	}
+	po := azqueue.PipelineOptions{
+		Retry: retry,
```
[the remark this answers is not in the document]
Comment (stuartnelson3) [reply module/apmazure/queue_test.go:42]: Unfortunately `azqueue.PipelineOptions` doesn't have `HTTPSender` :/

https://github.com/Azure/azure-storage-queue-go/blob/636801874cdd/azqueue/zc_pipeline.go#L8-L20

[the remark this answers is not in the document]
Comment (axw) [reply module/apmazure/queue_test.go:42]: ah I see. create a fake `pipeline.Pipeline` then? it's an interface

[the remark this answers is not in the document]
Comment (stuartnelson3) [reply module/apmazure/queue_test.go]: interesting choice on their part: https://github.com/Azure/azure-storage-queue-go/blob/636801874cdd/azqueue/zc_pipeline.go#L42

updated the tests to use the `fakeSender`.

**Comment by stuartnelson3** [comment]
jenkins run the tests

**Comment by stuartnelson3** [comment]
jenkins run the tests

**Pull request merged**